probe/tag/docker_tagger.go

```diff
@@ -217,88 +219,80 @@ func (m *dockerMapper) handleEvent(event *docker.APIEvents, client dockerClient)
 			return
 		}
 
-		m.Lock()
-		m.containers[containerID] = container
-		m.containersByPID[container.State.Pid] = container
-		m.Unlock()
+		t.Lock()
+		t.containers[containerID] = container
+		t.containersByPID[container.State.Pid] = container
+		t.Unlock()
 	}
 }
 
-func (m *dockerMapper) updatePIDTree() error {
-	pidTree, err := newPIDTreeStub(m.procRoot)
+func (t *dockerTagger) updatePIDTree() error {
+	pidTree, err := newPIDTreeStub(t.procRoot)
 	if err != nil {
 		return err
 	}
 
-	m.Lock()
-	m.pidTree = pidTree
-	m.Unlock()
+	t.Lock()
+	t.pidTree = pidTree
+	t.Unlock()
 	return nil
 }
 
-type dockerProcessMapper struct {
-	*dockerMapper
-	key string
-	f   func(*docker.Container) string
-}
-
-func (m *dockerProcessMapper) Key() string { return m.key }
-func (m *dockerProcessMapper) Map(pid uint) (string, error) {
-	var (
-		container *docker.Container
-		ok        bool
-		err       error
-		candidate = int(pid)
-	)
-
-	m.RLock()
-	for {
-		container, ok = m.containersByPID[candidate]
-		if ok {
-			break
-		}
-		candidate, err = m.pidTree.getParent(candidate)
-		if err != nil {
-			break
-		}
-	}
-	m.RUnlock()
-
-	if err != nil {
-		return "", fmt.Errorf("no container found for PID %d", pid)
-	}
+func (t *dockerTagger) Tag(r report.Report) report.Report {
```

Comment:
We seem to be mutating the passes in report here, although the intention seems to be for us not to do that.

If we are happy with the mutation, maybe make it a *Report and don't return Report.